src/Program.cs

```diff
@@ -1,29 +1,96 @@
 ﻿using ApiSdk;
 using Azure.Identity;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Graph.Cli.Authentication;
+using Microsoft.Graph.Cli.Commands.Authentication;
+using Microsoft.Graph.Cli.Configuration;
+using Microsoft.Graph.Cli.IO;
 using Microsoft.Graph.Cli.Utils;
 using Microsoft.Kiota.Authentication.Azure;
 using Microsoft.Kiota.Http.HttpClientLibrary;
 using System.CommandLine;
+using System.CommandLine.Builder;
+using System.CommandLine.Hosting;
+using System.CommandLine.Parsing;
+using System.IO;
+using System.Net.Http;
+using System.Reflection;
 using System.Threading.Tasks;
 
 namespace Microsoft.Graph.Cli
 {
-    class Program {
-        static async Task<int> Main(string[] args) {
-            DeviceCodeCredentialOptions credOptions = new()
-            {
-                ClientId = Constants.ClientId,
-                TenantId = Constants.TenantId
-            };
-
-            var credential = new DeviceCodeCredential(credOptions);
+    class Program
+    {
+        const string SdkVersionHeaderValueFormatString = "{0}-cli-{1}.{2}.{3}";
+
+        static async Task<int> Main(string[] args)
+        {
+            // We don't have access to a built host yet. Get configuration settings using a configuration builder.
+            // Required to set initial token credentials.
+            var configBuilder = new ConfigurationBuilder();
+            ConfigureAppConfiguration(configBuilder);
+            var config = configBuilder.Build();
+
+            var authSettings = config.GetRequiredSection(Constants.AuthenticationSection).Get<AuthenticationOptions>();
+            var authServiceFactory = new AuthenticationServiceFactory();
+            var authStrategy = AuthenticationStrategy.DeviceCode;
+
+            var credential = await authServiceFactory.GetTokenCredentialAsync(authStrategy, authSettings.TenantId, authSettings.ClientId);
             var authProvider = new AzureIdentityAuthenticationProvider(credential);
-            var core = new HttpClientRequestAdapter(authProvider);
+            var httpClient = new HttpClient();
+            var assemblyVersion = Assembly.GetExecutingAssembly().GetName().Version;
+            var sdkVersionHeaderValue = string.Format(
```

Comment:
look at the middleware collection and sync with @andrueastman on that aspect as I mentioned in the past. We already have the code for the injection in dotnet graph core (preview), I don't want to have this built twice for what is essentially the same ecosystem

Reply:
Created an issue for this. See #16